src/vm/typehandle.h

```diff
@@ -277,8 +277,12 @@ class TypeHandle
     // - do not restore encoded pointers on demand
     // - does not throw, does not trigger GC
     // - return type is three-valued (CanCast, CannotCast, MaybeCast)
-    // - MaybeCast indicates that the test tripped on an encoded pointer
-    //   so the caller should now call CanCastTo if it cares
+    //
+    // MaybeCast indicates an inconclusive result
+    // - the test tripped on an encoded pointer
```

Comment:
This comment about encoded pointer is wrong. The casting helpers did not handle encoded pointers.

Reply:
I think at some point encoded pointers were one of the reasons for `NoGC` to return `MayBe`. It is  up to whether the item is cached or not now, so yes, comments need updating. I have updated those that I found.